backend/telemetry_core/src/state/chain.rs

```diff
@@ -35,6 +36,298 @@ id_type! {
 pub type Label = Box<str>;
 
 const STALE_TIMEOUT: u64 = 2 * 60 * 1000; // 2 minutes
+const STATS_UPDATE_INTERVAL: Duration = Duration::from_secs(5);
+
+#[derive(Default)]
+struct Counter<K> {
+    map: HashMap<K, u64>,
+    empty: u64,
+}
+
+impl<K> Counter<K>
+where
+    K: Sized + std::hash::Hash + Eq,
+{
+    fn modify<'a, Q>(&mut self, key: Option<&'a Q>, increment: bool)
```

Comment:
I initially felt like having `increment` and `decrement` would be clearer, but I can sortof see why it's done this way. Instead of an `increment` boolean, how about an `enum CounterValue { Increment, Decrement }` for clarity?